bot/notification.py

```diff
@@ -0,0 +1,204 @@
+import logging
+
+from telegram import InputMediaPhoto
+from telegram.error import TelegramError
+
+from bot.bot import application
+from bot.models import TelegramIdAd
+from config import telegram_config
+from mongo import db
+
+from app.models import Ad
+
+
+chat_id = telegram_config.id_antalya_chat
+channel_id = telegram_config.id_antalya_channel
+
+logger = logging.getLogger(__name__)
+
+closed_ares = [area["name"] for area in db.areas.find({"is_closed": True})]
+connection_parameters = dict(connect_timeout=20, read_timeout=20)
+
+
+def check_floor(ad: Ad, parameter: list[str]) -> bool:
+    if "without_last" in parameter and ad.data.floor == str(ad.data.building_floor_count):
+        return False
+    if "without_first" in parameter and ad.data.floor in ["Elevation 1", "Garden-Floor"]:
+        return False
+    if "without_basement" in parameter and ad.data.floor in ["Basement", "Ground Floor", "Raised Ground Floor"]:
+        return False
+    return True
+
+
+def check_rooms(ad: Ad, parameter: list[str]) -> bool:
+    if "studio" in parameter and ad.data.room_count == "Studio Flat (1+0)":
+        return True
+    rooms, *_ = ad.data.room_count.split("+")
+    if "one" in parameter and rooms in ["1", "1.5"]:
+        return True
+    if "two" in parameter and rooms in ["2", "2.5"]:
+        return True
+    if "three" in parameter and rooms in ["3", "3.5+1"]:
+        return True
+    if "four" in parameter and rooms in ["4", "4.5", "5", "5.5", "6", "7", "8", "9", "10", "Over 10"]:
+        return True
```

Comment:
все эти списки с набором значений плохи тем, что легко допустить ошибку.
Например, в 41 строке ошибка, не будет в rooms никогда `3.5+1`
```suggestion
    rooms_str = ad.data.room_count
    rooms = "+" in s and int(float(rooms_str.split("+")[0]))
    if "one" in parameter and rooms == 1:
        return True
    if "two" in parameter and rooms == 2:
        return True
    if "three" in parameter and rooms == 3:
        return True
    if "four" in parameter and rooms  >= 4 or rooms_str == "Over 10":
        return True
```

Reply:
а еще если бы вместо `one` были бы сразу числа, то стало б еще проще. только придется оставить вариант `4+` и отдельно его обрабатывать 
 ```
    rooms_str = ad.data.room_count
    rooms = "+" in s and int(float(rooms_str.split("+")[0]))
    if "4+" in parameter and (rooms >= 4 or rooms_str == "Over 10"):
        return True
    elif rooms == int(parameter):
        return True
```
или даже IF-ы занимать на
```
   return any([
      rooms == int(parameter),
      "4+" in parameter and (rooms >= 4 or rooms_str == "Over 10"),
   ])
```

Reply:
последний вариант не подойдет, тогда там может False вылететь и логика сломается

Reply:
где будет False? у тебя метод и так False возвращает, если не срабатывает ни одна из веток IF. \
тут же any, а не all

Reply:
там в parameter лежит list

Reply:
типа так?
``` python
   return any([
      rooms in [int(p) for p in parameter if p.isnumeric()],
      "4+" in parameter and (rooms >= 4 or rooms_str == "Over 10"),
   ])
```

Reply:
вообще это дрочево конечно. вариант с IF более читаем, если он не сильно отличается количеству кода